content/PHPDataObjects/DataObjects.q.md

```diff
@@ -0,0 +1,1241 @@
+
+- `Male`
+
+- `John`
+
+- `Smith`
+
+- `first_name`
+
+- `1999-02-10`
+
+
+/// type=SS, answer=[5]
+
+In `student.php`, what does the query `INSERT INTO students (first_name, last_name, birth_date, gender) VALUES ('John', 'Smith', '1999-02-10', 'Male')` do on line 5?
+
+- It modifies multiple tables.
+
+- It adds three columns in the `students` table.
+
+- It inserts one column into the `students` table.
+
+- It modifies the values `John`, `Smith`, `1999-02-10`, and `Male` in the `students` table.
+
+- It adds a new row that contains the values `John`, `Smith`, `1999-02-10`, and `Male` in the `students` table.
+
+
+/// type=MS, answer=[2,3,4,5]
+
+Which statements correctly describe the code on line 5 of `student.php`?
+
+- It connects to the PostgreSQL database.
+
+- It calls the `query()` method of the `$conn` object.
+
+- It returns the result set as a `PDOStatement` object.
+
+- It assigns the `PDOStatement` object to the variable `$stmt`.
+
+- It executes the argument `INSERT INTO students (first_name, last_name, birth_date, gender) VALUES ('John', 'Smith', '1999-02-10', 'Male')` in the `query()` method.
+
+
+:::
+
+
+:::
+
+/// type=REPL, readonly=true, filename=[connection.php,student.php]
+
+```php
+// connection.php
+<?php
+    $host = 'localhost';
+    $db = 'LibraryDB';
+    $port = '5432';
+    $username = 'postgres';
+    $password = 'Admin01';
+
+    $dsn = "pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password";
+
+    try {
+        $conn = new PDO($dsn);
+        if ($conn) {
+            echo "Successfully connected to the database. ";
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection."; 
+    }
+?>
+```
+
+```php
+// student.php
+<?php
+    require_once("connection.php");
+
+    $stmt = $conn->query("SELECT * FROM students");
+    foreach ($stmt as $row) {
+        echo $row['student_id'].$row['first_name'].$row['last_name'].$row['birth_date'].$row['gender']. ' ';
+    }
+?>
+```
+/// type=SS, answer=[5]
+
+Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Unable to establish a connection.`.
+
+- It prints `Successfully connected to the database.` and `JohnSmith1999-10-02Male`.
+
+- It prints `Successfully connected to the database.` and `1JohnSmith1999-02-10Male`.
+
+
+/// type=MS, answer=[3,5]
+
+Which statements correctly describe the `foreach` construct on line 5 of `student.php`?
+
+- It removes all the elements in `$stmt`.
+
+- It accesses all the elements in `$stmt`.
+
+- It iterates through each element in the array.
+
+- It removes all the elements in the multidimensional array `$stmt`.
+
+- It assigns the value of the current element on each iteration to `$row`.
+
+
+/// type=SS, answer=[2]
+
+What value is assigned to `student_id`?
+
+- `0`
+
+- `1`
+
```

Comment:
Why `1`? I don't get the question.

Reply:
After running the query `select * from students` output is `1JohnSmith1999-02-10Male` so thus the question asks `What value is assigned to student_id?`
But, let me go ahead and rephrase the question. Thanks.